src/utils.py

```
@@ -8,122 +9,282 @@
+            logger.debug(f"Creating embeddings for {len(texts)} texts (attempt {retry + 1})")
             response = embedding_client.embeddings.create(
-                model=EMBEDDING_MODEL,
+                model=openai_config.embedding_model,
                 input=texts
             )
-            return [item.embedding for item in response.data]
+
+            # Ensure each embedding matches DB vector dimension (1024)
+            target_dim = 1024
+            def _fix_dim(vec: List[float]) -> List[float]:
+                if len(vec) == target_dim:
```

Ruff (0.11.9)

Function definition does not bind loop variable `target_dim`

(B023)